src/js/report.js

```diff
@@ -80,12 +82,21 @@ class Report {
 	}
 
 	updatePermalink() {
+		const url = new URL(this.baseUrl);
+		const lens = this.lens;
+
+		// TODO: Change subdomain.
```

Comment:
Would we update the subdomain *and* add the lens query param, or just the former?

Reply:
We would update the subdomain instead of the param.

Reply:
What's the TODO there, is it setting up DNS, or is it more involved? Could we setup a wildcard such that we don't have to manually add DNS records each time?

Reply:
There are two steps needed for adding a subdomain:

- register it on Google Domains
- register it with the GCP load balancer, which also enables TLS

I've already done these steps for the new [Drupal](drupal.httparchive.org) and [Magento](magento.httparchive.org) lenses. I'll look into ways to use wildcards to avoid the need for these steps for future lenses.

The TODO here is to parse the permalink URL and add/remove/replace the subdomain accordingly rather than relying on the `lens` URL param. The nice thing about the param is that it works out of the gate for all new lenses and there's already a supported [`searchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) API for add/remove/replace actions.